### PR TITLE
MDEV-14558: Indicating Maturity level of the Server

### PR DESCRIPTION
**Pull request by MohammedHMateen**
This pull request solves the ticket `MDEV-14558`, where I've tried to display the maturity of the server when queried with:
- `select @@version;` , `select version();` and related queries on the *client* side.
- `mysqld -V` , `mariadbd -V` and related queries on the *server* side.

### CODE REVIEW

**Comment by an3l** [comment]
Hi @MohammedHMateen,
Thanks for your contribution.

Similar to other open source projects, the MariaDB Foundation needs to have shared ownership of all code that is included in the MariaDB distribution. The easiest way to achieve this is by submitting your code under the BSD-new license.

The other alternative is to sign the code contribution agreement which can be found here: https://mariadb.com/kb/en/mariadb/mca/

Please indicate in a comment below, or update your PR comment, that you are contributing your new code of the whole pull request, including one or several files that are either new files or modified ones, under the BSD-new license or that you have filled out the contribution agreement and sent it.

Thanks,
Anel

File: sql/mysqld.cc [hunk 1]
```diff
@@ -8820,6 +8820,28 @@ static int get_options(int *argc_ptr, char ***argv_ptr)
   return 0;
 }
 
+/*
+  This function uses the maturity codes from _ and returns the respective maturity string as in
+  -unknown  -experimental -alpha -beta -gamma 
+  Note we do not display anything for stable versions*/
+  
+  const char* get_server_maturity_string(int maturity_code){
```
Comment (an3l) [inline sql/mysqld.cc:8828]: `{` should go in new line.

File: sql/mysqld.cc [hunk 2]
```diff
@@ -8820,6 +8820,28 @@ static int get_options(int *argc_ptr, char ***argv_ptr)
   return 0;
 }
 
+/*
+  This function uses the maturity codes from _ and returns the respective maturity string as in
```
Comment (an3l) [inline sql/mysqld.cc:8824]: We are trying to stick to 80 char rule, this line should have a break after ` respective`.

File: sql/mysqld.cc [hunk 3]
```diff
@@ -8820,6 +8820,28 @@ static int get_options(int *argc_ptr, char ***argv_ptr)
...
+  -unknown  -experimental -alpha -beta -gamma 
+  Note we do not display anything for stable versions*/
```
Comment (an3l) [inline sql/mysqld.cc:8826]: `*\` new line

File: mysql-test/main/server_maturity_in_version.test [hunk 4]
```diff
@@ -0,0 +1,5 @@
+--echo #
+--echo # MDEV - 14558: Indicate maturity level of the server.
+--echo #
+SELECT @@version;
```
Comment (grooverdan) [inline mysql-test/main/server_maturity_in_version.test:4]: So every version bump doesn't need to change this test, can you do `SELECT @@version RLIKE ....` (or another regex) to make the expression just return 1 if successful. Same with `SELECT version()` below.

File: sql/mysqld.cc [hunk 5]
```diff
@@ -8820,6 +8820,28 @@ static int get_options(int *argc_ptr, char ***argv_ptr)
...
+            break;
+    case 1: return "-experimental";
+            break;  
+    case 2: return "-alpha";
+            break;
+    case 3: return "-beta";
+            break;
+    case 4: return "-gamma";
+            break;
+    default:return NULL;
```
Comment (grooverdan) [inline sql/mysqld.cc:8841]: NULL will get interpreted as the last part when consumed by `strxnmov` so maybe `""` here (on a new line too).

Comment (grooverdan) [reply sql/mysqld.cc:8841]: could remove the `SERVER_MATURITY_LEVEL` fixed argument too and `compile_assert` to ensure the prescribed level SERVER_MATURITY_LEVEL hasn't been altered.

File: mysql-test/main/server_maturity_in_version.test (hunk 4 above)
Comment (grooverdan) [reply mysql-test/main/server_maturity_in_version.test:4]: the regex also needs to account this may not be a debug build.

**Pull request closed without merging**